### DISCUSSION TITLE
Reverse proxy HTTPS

### DISCUSSION THREAD

**Discussion by Aerglonus** [tobychui/zoraxy#220] [Q&A] [answered]
I recently been trying Zoraxy but can't find a way to make it reverse proxy to an HTTPS server for example in NPM i can add it like this
(this example is for passbolt https server)
![image](https://github.com/tobychui/zoraxy/assets/1855681/5abebb2a-0f56-40e6-9008-137a85fc40f9)
and this will use the https server instead, in Zoraxy i tried enabling the `Require TLS` option but it returns 404 not found when visiting the url. 

is there a way to do something like in NPM or is this not supported in Zoraxy yet?

**Comment by tobychui** [comment]
I guess you have some invalid setting somewhere. Screenshot your zoraxy settings and put it here, otherwise there are nothing we can help.

**Reply by Aerglonus** [reply]
Sure this is the proxy record (in this screenshot Require TLS is disabled and opens the usual `400` request but when enabled it returns 404 not found)
![Screenshot 2024-06-26 193755](https://github.com/tobychui/zoraxy/assets/1855681/beceb89e-5b2f-4284-8dac-b0013172a505)
With `Require TLS` Enabled 
![image](https://github.com/tobychui/zoraxy/assets/1855681/66484518-a6e7-4b32-b80f-793b5fa7ceae)


And here is the Global Settings 
![image](https://github.com/tobychui/zoraxy/assets/1855681/8889a28f-cef0-4fec-8bc8-41d60e3b2224)

Nothing other than the added certificates is changed from the settings.

**Reply by tobychui** [reply, marked as answer]
@Aerglonus There are a few reasons for it. First of all, the 404 not found is probably not from Zoraxy but your proxied services. For your information, 404 page in Zoraxy looks like this.
![圖片](https://github.com/tobychui/zoraxy/assets/24617523/41831258-c9c2-4d26-be1e-52a9b8a5138a)

Secondly, do your backend server (.1.65) used a self-sign certificate for the https? If yes, you will also need to enable Skip Verification. Otherwise, I would usually recommend proxying via HTTP if both of your nodes are within your private / local area network.

Thirdly, I assume that the 400 error does not origin from Zoraxy but your backend / other services. (Zoraxy will only shows error of 500, 521 and 404 as I remembered). 

So I guess you really need to check your networking and config to see what external services is causing trouble in your routing setup. My best guess is that in order to make your origin server compatible with NPM (which is buggy and need a lot of hack to make them works together), you have modified some configs from your origin server. You might want to undo them if that is the case.

**Reply by Aerglonus** [reply]
`need to enable Skip Verification`
This was what i needed to do, i don't what was causing the 404 page but enabling this worked. ( to disabled it i just needed to force SSL on passbolt side.)

Thanks
